openstack/placement/resourceproviders/doc.go

```diff
@@ -0,0 +1,21 @@
+/*
+Package resourceproviders lists all resource providers from the OpenStack Placement service.
+
+Example to list resource providers
+
+	allPages, err := resourceproviders.List(placementClient, resourceproviders.ListOpts{}).AllPages()
+    if err != nil {
+        panic(err)
+    }
+
+	allResourceProviders, err := resourceproviders.ExtractResourceProviders(allPages)
+    if err != nil {
+        panic(err)
+    }
+
+	for _, r := range allResourceProviders {
+        fmt.Printf("%+v\n", r)
+	}
```

Comment:
nit: the formatting of the above example code looks a little off. godoc depends on correct formatting or else it wouldn't be a big deal.

Reply:
You are right.  I'll fix that.